### PR TITLE
refactor (rs-flowershop-121) Project refactoring

### PR DESCRIPTION
**Pull request by shchegolenkov**
## Pull request type

- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Bugfix
- [x] Build related changes
- [ ] Documentation update
- [ ] Tests
- [ ] Other

## What's new?

- Optimize redux actions in components
- Improve conditional classnames syntax
- Use function expression syntax
- Improve css-rules order
- Audit dependencies
- Improve import order
- Apply path aliases
- Improve code readability, naming

## Related Issues

#121 

## Checklist before merge

- [x] This pull request merges to the current sprint branch
- [x] I have reviewed the code for potential issues.
- [x] I have fixed any warnings or errors
- [ ] Approved by team developers

### CODE REVIEW

File: src/layouts/RootLayout/Footer/index.tsx [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 
-import LogoIco from '@/assets/svg/logo.svg';
+import { RouteLinks } from '@/types/types';
+
+import LogoIcon from '@/assets/svg/logo.svg';
 
 import s from './Footer.module.scss';
 
 const links = [
-  { to: '/catalog', text: 'All products' },
```
Comment (Mialiokhin) [inline src/layouts/RootLayout/Footer/index.tsx:9]: вместо text лучше назвать title

File: src/pages/LoginPage/index.tsx [hunk 2]
```diff
@@ -1,66 +1,49 @@
...
-      .min(8, 'Password must be at least 8 characters')
-      .matches(/[A-Z]/, 'Password must contain at least one uppercase letter (e.g., [A-Z])')
-      .matches(/[a-z]/, 'Password must contain at least one lowercase letter (e.g., [a-z])')
-      .matches(/[0-9]/, 'Password must contain at least one digit (e.g., [0-9])')
-      .matches(
-        /[!@#$%^&*]/,
-        'Password must contain at least one special character (e.g., !@#$%^&*)'
-      )
-      .matches(/^\S*$/, 'Password cannot contain spaces'),
+    email: emailRules,
```
Comment (Mialiokhin) [inline src/pages/LoginPage/index.tsx]: emailSchema лучше переименовать и тд

File: src/pages/LoginPage/index.tsx [hunk 3]
```diff
@@ -118,32 +104,30 @@ const LoginPage: React.FC = () => {
...
-                ) : null}
-                {message ? (
-                  <Alert variant="outlined" severity="error">
-                    {message}
-                  </Alert>
-                ) : null}
-              </form>
-            </ThemeProvider>
+            <form className={s.form} onSubmit={handleSubmit(onSubmit)}>
+              <EmailInput register={register} errors={errors} />
```
Comment (Mialiokhin) [inline src/pages/LoginPage/index.tsx:108]: Инпуты лучше переделать как <Input type='email' и тд

File: src/pages/ProfilePage/NewAddressBlock/index.tsx [hunk 4]
```diff
@@ -16,58 +17,48 @@ import {
...
-import FormTheme from '@/themes/FormTheme';
+import { COUNTRIES } from '@/constants';
 import { AddressAction, ApiResponse, ProfileForm, Status, User } from '@/types/types';
 
 import s from '../ProfilePage.module.scss';
 
-interface ProfileEditBlockProps {
+const getAddressName = (typeAddress: string) => {
+  let message = '';
+  if (typeAddress === 'shipping') {
```
Comment (Mialiokhin) [inline src/pages/ProfilePage/NewAddressBlock/index.tsx:34]: лучше типизировать 'shipping' и 'billing'

File: src/pages/ProfilePage/ProfileAddressBlock/addressValidation.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import * as yup from 'yup';
+
+export const schema = yup.object().shape({
```
Comment (Mialiokhin) [inline src/pages/ProfilePage/ProfileAddressBlock/addressValidation.ts]: может собрать все схемы валидации сайта во один файл?

File: src/pages/ProfilePage/index.tsx [hunk 6]
```diff
@@ -231,71 +204,69 @@ const ProfilePage: React.FC = () => {
...
+              register={register}
+              errors={errors}
+              control={control}
+              reset={reset}
+              defaultValue={user && user.dateOfBirth ? new Date(user.dateOfBirth) : null}
+              isEditField={true}
+              isDisabled={isDisabledDateOfBirth}
+              switchEditModeField={handleDateOfBirthClick}
+            />
+            {!isDisabledFirstName ||
```
Comment (Mialiokhin) [inline src/pages/ProfilePage/index.tsx]: наверно лучше собрать условия в одну константу и дать локоничное название

**Review by Mialiokhin — APPROVED** [review]

File: src/pages/ProfilePage/ProfileAddressBlock/addressValidation.ts (hunk 5 above)
Comment (shchegolenkov) [reply src/pages/ProfilePage/ProfileAddressBlock/addressValidation.ts]: Перенёс в общий файл со схемами валидации https://github.com/shchegolenkov/rs-flowershop/commit/6b45a63c9af78d4442abe96519774626fb2333b1

File: src/pages/ProfilePage/NewAddressBlock/index.tsx (hunk 4 above)
Comment (shchegolenkov) [reply src/pages/ProfilePage/NewAddressBlock/index.tsx:34]: done https://github.com/shchegolenkov/rs-flowershop/commit/0f78ef737a2d301994e88d1c54da329d11145d0f

File: src/pages/LoginPage/index.tsx (hunk 3 above)
Comment (shchegolenkov) [reply src/pages/LoginPage/index.tsx:108]: done https://github.com/shchegolenkov/rs-flowershop/commit/e6433df0c060c81a7f26a3a581a4d69dbb1f3185

File: src/layouts/RootLayout/Footer/index.tsx (hunk 1 above)
Comment (shchegolenkov) [reply src/layouts/RootLayout/Footer/index.tsx:9]: done https://github.com/shchegolenkov/rs-flowershop/commit/79503bf27f51cf7502876748819b320847a8df7c

File: src/pages/LoginPage/index.tsx (hunk 2 above)
Comment (shchegolenkov) [reply src/pages/LoginPage/index.tsx]: done https://github.com/shchegolenkov/rs-flowershop/commit/79503bf27f51cf7502876748819b320847a8df7c

File: src/pages/ProfilePage/index.tsx (hunk 6 above)
Comment (shchegolenkov) [reply src/pages/ProfilePage/index.tsx]: done https://github.com/shchegolenkov/rs-flowershop/commit/79503bf27f51cf7502876748819b320847a8df7c

**Review by Mialiokhin — APPROVED** [review]

**Pull request merged**